sdk/src/Services/DynamoDBv2/Custom/DataModel/IDynamoDBContext.cs

```diff
@@ -156,7 +199,32 @@ public partial interface IDynamoDBContext : IDisposable
         /// <param name="valuesType">The type of data which will be persisted in this batch.</param>
         /// <param name="operationConfig">Config object which can be used to override that table used.</param>
         /// <returns>Empty strongly-typed BatchWrite object</returns>
-        BatchWrite<object> CreateBatchWrite(Type valuesType, DynamoDBOperationConfig operationConfig = null);
+        [Obsolete("Use the CreateBatchWrite overload that takes BatchWriteConfig instead, since DynamoDBOperationConfig contains properties that are not applicable to BatchWrite.")]
+        BatchWrite<object> CreateBatchWrite(Type valuesType, DynamoDBOperationConfig operationConfig);
```

Comment:
why did you change the signature of the method?

Reply:
See https://github.com/aws/aws-sdk-net/pull/3376#discussion_r1672274431